lib/ace/transport_app.rb

```diff
@@ -89,32 +93,40 @@ def validate_schema(schema, body)
     post '/execute_catalog' do
       content_type :json
 
+      # TODO: catch any parser errors and handle with JSON response
       body = JSON.parse(request.body.read)
       error = validate_schema(@schemas["execute_catalog"], body)
       return [400, error.to_json] unless error.nil?
 
-      # simulate expected error cases
-      if body['compiler']['certname'] == 'fail.example.net'
-        [200, { _error: {
-          msg: 'catalog compile failed',
-          kind: 'puppetlabs/ace/compile_failed',
-          details: 'upstream api errors go here'
-        } }.to_json]
-      elsif body['compiler']['certname'] == 'credentials.example.net'
-        [200, { _error: {
-          msg: 'target specification invalid',
-          kind: 'puppetlabs/ace/target_spec',
-          details: 'upstream api errors go here'
-        } }.to_json]
-      elsif body['compiler']['certname'] == 'reports.example.net'
-        [200, { _error: {
-          msg: 'report submission failed',
-          kind: 'puppetlabs/ace/reporting_failed',
-          details: 'upstream api errors go here'
-        } }.to_json]
-      else
-        [200, '{}']
-      end
+      # TODO: perform validation on body receieved, return an error
+
+      # TODO: retrieve these from the target
+      facts = {}
+      trusted_facts = {}
+
+      http_response = @client.retrieve(
```

Comment:
There is no need to hook up the `catalog.retrieve` to the `transport_app`. The actuall call-stack will look very different anyways.

A solid test coverage of the `Catalog` class itself is all that's needed.

Reply:
This has all been removed and the mock endpoints restored